libsolidity/formal/SMTEncoder.cpp

```diff
@@ -391,11 +391,16 @@ void SMTEncoder::endVisit(TupleExpression const& _tuple)
 	createExpr(_tuple);
 
 	if (_tuple.isInlineArray())
-		m_errorReporter.warning(
-			2177_error,
-			_tuple.location(),
-			"Assertion checker does not yet implement inline arrays."
-		);
+	{
+		// Add constraints for the length and values as it is known.
+		auto symbArray = dynamic_pointer_cast<smt::SymbolicArrayVariable>(m_context.expression(_tuple));
+		solAssert(symbArray, "");
+
+		vector<smtutil::Expression> elementValues;
+		for (auto const& component: _tuple.components())
+			elementValues.push_back(expr(*component));
+		addArrayLiteralAssertions(*symbArray, elementValues);
```

Comment:
```suggestion
		addArrayLiteralAssertions(*symbArray, applyMap(_tuple.components(), [&](auto const& c) { return expr(*c); }));
```
Could do something like that instead if preferred, but I'm fine with. If changes, this could also be applied to the other application of `addArrayLiteralAssertions` as well.

Reply:
I'll change this now